lib/bolt/config/modulepath.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'bolt/config'
+
+module Bolt
+  class Config
+    class Modulepath
+      BOLTLIB_PATH = File.expand_path('../../../bolt-modules', __dir__)
+      MODULES_PATH = File.expand_path('../../../modules', __dir__)
+
+      def initialize(modulepath)
+        @modulepath = if modulepath.nil?
+                        []
+                      elsif modulepath.is_a?(String)
+                        [modulepath]
+                      else
+                        modulepath
+                      end
+      end
+
+      # The bolt_compiler_modulepath includes both the BOLTLIB
+      # path and the MODULES_PATH to ensure bolt functions and
+      # built-in content are available in the compliler
+      def bolt_compiler_modulepath
+        [BOLTLIB_PATH, *@modulepath, MODULES_PATH]
+      end
+
+      # The apply_compiler_modulepath only includes the original
+      # modulepath, since we don't want to pluginsync any of the
+      # content from BOLT_MODULES since that content includes
+      # core modules that can conflict with modules installed
+      # with an agent
+      def apply_compiler_modulepath
```

Comment:
This one is a little weird. It's really more like `plugin_modules`, since actual _compilation_ of apply blocks uses the entire modulepath. It's only when gathering facts and when applying the catalog that we use a subset of them. But if we had a `user_modules` method then maybe that's what we would want to use here.